source/particle/generator/interface.cc

```diff
@@ -114,6 +114,24 @@ namespace aspect
               }
             ++iteration;
           }
+        // If the maximum iterations are reached, retry generating particles
+        // within the reference cell
+        if (iteration == maximum_iterations)
+          {
+            iteration = 0;
```

Comment:
You don't need this if you delete the assert below. It will never trigger anyways...